index.txt

```diff
@@ -60,110 +60,110 @@ install_caddy()
 
 	# NOTE: `uname -m` is more accurate and universal than `arch`
 	# See https://en.wikipedia.org/wiki/Uname
-	if [ -n "$(uname -m | grep aarch64)" ]; then
+	unamem=$(uname -m)
+	if grep -q aarch64 <<<"$unamem"; then
```

Comment:
That is making a string into a (virtual) file, it's called a Here string. Because grep works on file arguments, but we don't need a file, we don't have a file, we don't want a file, we want to grep through a string. We don't need to pipe (making an extra process and other pipe complications), this is the 'cheapest' and most elegant pattern.